src/analyses/mutexAnalysis.ml

```diff
@@ -114,39 +159,36 @@ struct
       `Index (i_exp, conv_offset_inv o)
 
   let query ctx (type a) (q: a Queries.t): a Queries.result =
-    let check_fun ~write ls =
-      let locks = Lockset.export_locks ls in
-      if write then (Mutexes.bot (), locks) else (locks, Mutexes.bot ())
-    in
-    let non_overlapping locks1 locks2 =
-      let intersect = GProtecting.join locks1 locks2 in
-      GProtecting.is_top intersect
-    in
+    (* get the set of mutexes protecting the variable v in the given mode *)
+    let protecting ~write mode v = GProtecting.get ~write mode (G.protecting (ctx.global (V.protecting v))) in
+    let non_overlapping locks1 locks2 = Mutexes.is_empty @@ Mutexes.inter locks1 locks2 in
```

Comment:
This changes the `non_overlapping` check back to just `Mutexes` operations. For traces in 1ce6e7889419f8e96acc551537752ea485b23ebc, it was changed from `Mutexes` to the global invariant based check and there was some reason for making the check more abstract.

Are you sure that the `GProtecting` version isn't still needed? It might have to do with the read- and write-lock distinction, which is not captured by `Mutexes`. Could be that a regression test was never added for that and this is making it less precise again.

Reply:
I don't think it is necessary here. All values passed to `non-overlapping` were constructed by `check_fun` such that there is only one component that is not the empty lockset. So I think the new code is functionally equivalent but clearer to read now!

Reply:
Looks like originally it was for OSEK mutexes and their priorities. In the remaining instance `G` was referring to the same type of locks, so indeed it is equivalent.